### DISCUSSION TITLE
double-tap zoom problem

### DISCUSSION THREAD

**Discussion by Ovoron** [zauberzeug/nicegui#3238] [Q&A] [answered]
### Question

Hello @all
I have the problem that after repeatedly or quickly pressing a button on my smartphone, the area is zoomed. 
How can I deactivate this? 
double-tap zoom problem
```py
ui.run(
    title='Meine App von AS',
    reload=False,
    port=8082,
    binding_refresh_interval=0.2,
    show=False,
    viewport='width=device-width, height=device-height, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no',
)
```
I have also read that this can be deactivated with css. 
```css
touch-action: manipulation;
```

What is the best way to disable the zoom function ?
best regards
Ovoron

**Comment by falkoschindler** [comment, marked as answer]
Hi @Ovoron,

I had the same problem with a personal project. Setting `touch-action: manipulation` for all buttons worked in my case:
```py
ui.button.default_classes('touch-manipulation')  # NOTE: disable double-tap zoom on mobile devices
```

**Reply by Ovoron** [reply]
Thank You Falko,
that was the solution !!

I have another question.
I want to create 3 buttons in a row.
Every button with the same size and across the whole width.

```
with ui.row().style('width: 100%; display: flex;'):
        ui.button('Button 1').style('flex: 1; margin: 5px;')
        ui.button('Button 2').style('flex: 1; margin: 5px;')
        ui.button('Button 3').style('flex: 1; margin: 5px;')
#or
with ui.row().classes("justify-center w-full border"):
        ui.button('Button 1')
        ui.space()
        ui.button('Button 2')
        ui.space()
        ui.button('Button 3')
```

The problem is that not the entire width is used. 
What do I need to add?

I have attached a picture of this problem.

Can you help me again?

Greetings Ovoron

![Tetris_Problem_2nd](https://github.com/zauberzeug/nicegui/assets/149179724/98a9d34c-99ed-4e3e-8d78-efb782a1e346)

**Reply by rodja** [reply]
Please start a new Q&A for each separate topic.
